### PR TITLE
LibGfx: Add scaffolding for a webp decoder

### PR DESCRIPTION
**Pull request by nico**
At the moment, this processes the RIFF chunk structure and extracts the ICCP chunk, so that `icc` can now print ICC profiles embedded in webp files. (And are image files really more than containers of icc profiles?)

It doesn't even decode image dimensions yet.

The lossy format is a VP8 video frame. Once we get to that, we might want to move all the image decoders into a new LibImageDecoders that depends on both LibGfx and LibVideo. (Other newer image formats like heic and av1f also use video frames for image data.)

### CODE REVIEW

File: Userland/Utilities/file.cpp [hunk 1]
```diff
@@ -122,7 +122,7 @@ static Optional<DeprecatedString> elf_details(DeprecatedString description, Depr
     __ENUMERATE_MIME_TYPE_DESCRIPTION("image/bmp", "BMP image data", image_details)                                 \
     __ENUMERATE_MIME_TYPE_DESCRIPTION("image/gif", "GIF image data", image_details)                                 \
     __ENUMERATE_MIME_TYPE_DESCRIPTION("image/jpeg", "JPEG image data", image_details)                               \
-    __ENUMERATE_MIME_TYPE_DESCRIPTION("image/png", "PNG image data", image_details)                                 \
+    __ENUMERATE_MIME_TYPE_DESCRIPTION("image/webp", "WebP image data", image_details)                               \
```
Comment (Hendiadyoin1) [inline Userland/Utilities/file.cpp:125]: Removing PNG in favor of partial webp is a bit overeager

Comment (nico) [reply Userland/Utilities/file.cpp:125]: :yakkie:

File: Userland/Libraries/LibGfx/WebPLoader.cpp [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+    ReadonlyBytes data;
+
+    RefPtr<Gfx::Bitmap> bitmap;
+
+    Optional<ReadonlyBytes> icc_data;
+};
+
+static ErrorOr<void> decode_webp_header(WebPLoadingContext& context)
+{
+    // https://developers.google.com/speed/webp/docs/riff_container#webp_file_header
```
Comment (linusg) [inline Userland/Libraries/LibGfx/WebPLoader.cpp:75]: Please put spec links above the function if it applies to the entire body (LibWeb/LibJS style)

File: Userland/Libraries/LibGfx/WebPLoader.cpp [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+
+    if (first_chunk.type == FourCC("VP8X"))
+        return decode_webp_extended(context, first_chunk, chunks);
+
+    return Error::from_string_literal("WebPImageDecoderPlugin: Invalid first chunk type");
+}
+
+WebPImageDecoderPlugin::WebPImageDecoderPlugin(ReadonlyBytes data, OwnPtr<WebPLoadingContext> context)
+{
+    m_context = move(context);
```
Comment (linusg) [inline Userland/Libraries/LibGfx/WebPLoader.cpp:213]: I don't know what it's called, but we prefer this style in ctors:

```suggestion
    : m_context(move(context))
{
```

File: Userland/Utilities/file.cpp [hunk 4]
```diff
@@ -123,6 +123,7 @@ static Optional<DeprecatedString> elf_details(DeprecatedString description, Depr
     __ENUMERATE_MIME_TYPE_DESCRIPTION("image/gif", "GIF image data", image_details)                                 \
     __ENUMERATE_MIME_TYPE_DESCRIPTION("image/jpeg", "JPEG image data", image_details)                               \
     __ENUMERATE_MIME_TYPE_DESCRIPTION("image/png", "PNG image data", image_details)                                 \
+    __ENUMERATE_MIME_TYPE_DESCRIPTION("image/webp", "WebP image data", image_details)                               \
```
Comment (linusg) [inline Userland/Utilities/file.cpp:126]: This is not really related to the LibGfx change and should be its own commit :)

**Review by linusg — CHANGES_REQUESTED** [review]
Looks good, some nitpicks

File: Userland/Libraries/LibGfx/WebPLoader.cpp (hunk 3 above)
Comment (nico) [reply Userland/Libraries/LibGfx/WebPLoader.cpp:213]: That's called an initializer list. Done. (In some earlier version of this patch that wasn't possible, and then I forgot to update it when it became possible.)
